packages/analytics-browser/src/browser-client.ts

```diff
@@ -50,8 +50,15 @@ export class AmplitudeBrowser extends AmplitudeCore<BrowserConfig> {
     await this.add(new Context());
     await this.add(new Destination());
 
-    // Step 5: Track attributions
+    // Step 5: Set timeline ready for processing events
+    this.timeline.isReady = true;
+
+    // Step 6: Track attributions
     await this.runAttributionStrategy(options?.attribution, isNewSession);
+
+    // Step 7: Flush existing events, which might be collected by track before init
+    // This flush needs to run after plugin installation to gain the correct attributes
+    await this.timeline.flush();
```

Comment:
```suggestion
    // Step 5: Set timeline ready for processing events
    this.timeline.isReady = true;
    if (!this.config.optOut) {
        await this.timeline.scheduleApply();
    }

    // Step 6: Track attributions
    await this.runAttributionStrategy(options?.attribution, isNewSession);
```

Reply:
given the other suggestions i think we may need to do this here.

1. `config.optOut` check since we did not check this before init
2. Call scheduleApply instead to go through normal flow